data-protocols/ids/ids-api-multipart-dispatcher-v1/src/main/java/org/eclipse/dataspaceconnector/ids/api/multipart/dispatcher/sender/MultipartArtifactRequestSender.java

```diff
@@ -112,39 +120,48 @@ protected Message buildMessageHeader(DataRequest request, DynamicAttributeToken
         request.getProperties().forEach(message::setProperty);
         return message;
     }
-
+    
+    /**
+     * Builds the payload for the artifact request. The payload contains the data destination and a secret key.
+     *
+     * @param request the request.
+     * @return the message payload.
+     * @throws Exception if parsing the payload fails.
+     */
     @Override
     protected String buildMessagePayload(DataRequest request) throws Exception {
 
-        ArtifactRequestMessagePayload.Builder requestPayloadBuilder = ArtifactRequestMessagePayload.Builder.newInstance()
+        var requestPayloadBuilder = ArtifactRequestMessagePayload.Builder.newInstance()
                 .dataDestination(request.getDataDestination());
 
         if (request.getDataDestination().getKeyName() != null) {
             String secret = vault.resolveSecret(request.getDataDestination().getKeyName());
             requestPayloadBuilder = requestPayloadBuilder.secret(secret);
         }
-
-        ObjectMapper objectMapper = getObjectMapper();
-        return objectMapper.writeValueAsString(requestPayloadBuilder.build());
+        
+        return getObjectMapper().writeValueAsString(requestPayloadBuilder.build());
     }
-
+    
+    /**
+     * Parses the response content.
+     *
+     * @param parts container object for response header and payload input streams.
+     * @return a MultipartResponse containing the message header and the response payload as string.
+     * @throws Exception if parsing header or payload fails or if the response header is not of type
+     *                   RequestInProcessMessage.
+     */
     @Override
-    protected MultipartRequestInProcessResponse getResponseContent(IdsMultipartParts parts) throws Exception {
-        Message header = getObjectMapper().readValue(parts.getHeader(), Message.class);
+    protected MultipartResponse<String> getResponseContent(IdsMultipartParts parts) throws Exception {
+        var header = getObjectMapper().readValue(parts.getHeader(), Message.class);
         String payload = null;
         if (parts.getPayload() != null) {
             payload = new String(parts.getPayload().readAllBytes());
         }
 
-        if (header instanceof RequestInProcessMessage) {
-            // TODO Update TransferProcess State Machine
-        } else {
-            // TODO Update TransferProcess State Machine
+        if (header instanceof RejectionMessage) {
```

Comment:
Here the `ResponseUtil` could be used, I see that probably you didn't because this "validation", in my opinion the validation should be done in another place, after the return of the `getResponseContent`, like, for e particular type of request there should be a particular type of response.
I don't know if this is the right PR to address this, but we should think about it.

Reply:
replaced. Added a `checkResponseType` method that will run after the response of all senders - not only the artifact request.